purchasing_test/unit/opportunities/test_opportunity_admin.py

```diff
@@ -423,6 +423,18 @@ def test_pending_opportunity_admin(self):
         self.assertEquals(admin_publish.status_code, 302)
         self.assertTrue(Opportunity.query.get(self.opportunity3.id).is_public)
 
+    def test_approved_opportunity(self):
+        '''Test approved opportunities work as expected for city staff
+        '''
+        self.login_user(self.admin)
+        admin_publish = self.client.get('/beacon/admin/opportunities/{}/publish'.format(
+            self.opportunity1.id
+        ))
+        self.assert_flashes('Opportunity successfully published!', 'alert-success')
+        self.assertEquals(admin_publish.status_code, 302)
+        self.assertTrue(Opportunity.query.get(self.opportunity1.id).is_public)
+        self.assertFalse(Opportunity.query.get(self.opportunity1.id).is_advertised)
```

Comment:
You'll want to add the test we talked about -- logged out users shouldn't be able to get to the detail page for this opportunity, but staff should be able to.

```
self.assert200(self.client.get('/beacon/opportunities/{}'.format(self.opportunity1.id))
self.logout_user()
self.assert404(self.client.get('/beacon/opportunities/{}'.format(self.opportunity1.id))
```

or something along those lines